### PR TITLE
Add pandoc support

### PR DESCRIPTION
**Pull request by jmuheim**
I need Pandoc in my Rails project. This is my attempt to register Pandoc as markdown parser in Tilt.

Any improvements are very welcome.

Here's the original question which made me create this PR: [Slim template: use Pandoc as markdown converter?](http://stackoverflow.com/questions/30058230/slim-template-use-pandoc-as-markdown-converter)

The tests aren't all passing yet. In fact, I'm a bit unsure on how to proceed, because of the following questions:

- As the used `pandoc-ruby` gem is only a wrapper for the `$ pandoc` command line tool, I'm unsure whether somehow it should be checked whether pandoc is installed on the system or not. Maybe `pandoc-ruby` does this already?
- Pandoc has a huge amount of special features. For examples, I love the [footnotes](http://pandoc.org/README.html#footnotes). Should some of these things be tested, too? I'm asking because I see there are some tests verifying that e.g. escaping HTML works ([`test_escape_html_true`](https://github.com/rtomayko/tilt/blob/master/test/tilt_markdown_test.rb#L31)).
- Some of the features are called a bit different in Pandoc: for example, the `:smartypants` option is only `:smart`.
    - Should I simply undef the default test?
    - Should I add a mechanism in `PandocTemplate` that converts the `:smartypants` option into `:smart`?

### CODE REVIEW

**Comment by judofyr** [comment]
This looks mostly fine :+1: 

* Checking for the `pandoc` binary isn't required IMO
* As for features: If you're registering under `.mkd, .markdown` it would be nice if the same set of options work across all implementation. As you can see from the tests, it's quite difficult maintaining this parity.
* One option could be to introduce a subclass `PandocMarkdownTemplate` which maps the options, and keep `PandocTemplate` unregistered.

File: Gemfile [hunk 1]
```diff
@@ -36,6 +36,7 @@ group :secondary do
   gem 'asciidoctor', '>= 0.1.0'
   gem 'liquid'
   gem 'maruku'
+  gem 'pandoc-ruby'
```
Comment (jmuheim) [inline Gemfile]: Don't know whether this should be in group primary or secondary.

Comment (jmuheim) [inline Gemfile]: Don't even know whether it's a good idea to load pandoc-ruby by default.

**Comment by jmuheim** [comment]
@judofyr 

> As for features: If you're registering under .mkd, .markdown it would be nice if the same set of options work across all implementation. As you can see from the tests, it's quite difficult maintaining this parity.

I have implemented the `:smartypants` option. I couldn't find a Pandoc option for `:escape_html` yet, test is set to `skip` at the moment. See http://stackoverflow.com/questions/37165374/pandoc-escape-html-option.

> One option could be to introduce a subclass PandocMarkdownTemplate which maps the options, and keep PandocTemplate unregistered.

I don't understand what you mean with that. Is it still relevant?

**Comment by jmuheim** [comment]
I've been able make the `:escape_html` switch work.

Please review this PR so it can be merged. Thank you.

**Comment by judofyr** [comment]
This looks very good! I'll wait for the build to pass, and then it's good to go.

**Comment by jmuheim** [comment]
Wait a sec, I have another commit.

**Comment by judofyr** [comment]
No problem. If you comment again after the build is green and everything is fine, I'll merge it :-)

File: Gemfile (hunk 1 above)
Comment (jmuheim) [inline Gemfile]: Is this okay to you? Or is there a way to only pandoc functionality when the user has `pandoc-ruby` installed already?

The problem then is: how to run the tests? I don't see a `:test` group in your Gemfile... I think it's not good to load so many gems that aren't needed (I don't need to have different markdown engines available, I always only use one).

Comment (judofyr) [inline Gemfile]: The Gemfile is only used for development. When you use the gem, no other markdown engines are loaded (and there's no other dependencies either).

**Comment by jmuheim** [comment]
Perfect. Please merge and deploy a new gem version. 👍

**Comment by judofyr** [comment]
2.0.3 has been released.

**Comment by jmuheim** [comment]
Yayyy! 😃

**Comment by judofyr** [comment]
Thanks for the patch! Very nicely done 👍

**Comment by jmuheim** [comment]
Damn, I forgot to remove the stuff from the README that I wrote when I thought that Pandoc wouldn't support escaping HTML. Here's a patch to change the two lines, maybe you can apply it directly to master and release another new version?

[readme_patch.diff.txt](https://github.com/rtomayko/tilt/files/261170/readme_patch.diff.txt)

**Comment by judofyr** [comment]
I've pushed it to master now. Not sure if it needs another release. It's just a very minor documentation issue.

**Comment by jmuheim** [comment]
That's right. I leave it up to you.

**Pull request merged**